vllm/entrypoints/openai/api_server.py

```diff
@@ -391,11 +416,27 @@ async def completion_full_generator():
         assert final_res is not None
 
         choices = []
+        prompt_token_ids = final_res.prompt_token_ids
+        prompt_logprobs = final_res.prompt_logprobs
         role = get_role()
         for output in final_res.outputs:
+            if request.logprobs is not None:
```

Comment:
if `request.logprobs` is False(or 0), it will still trigger this branch. Maybe a better way:

```
if request.logprobs:
    # do things
else:
   # do things
```

Please also update the same issue in other places.